docs/Reference-for-template.json.md

```diff
@@ -135,17 +135,20 @@ A symbol for which the config provides literal and/or default values.
-|`defaultValue`|The value assigned to the symbol if no parameter is provided by the user or host.|
+|<a name="defaultValue"></a>`defaultValue`|The value assigned to the symbol if no parameter is provided by the user or host. The default is *not applied* if [`isRequired`](#isRequired) configuration is set to `true` for a parameter (or [`isRequiredCondition`](#isRequiredCondition) is specified and evals to `true`), as that is an indication that user specified value is mandatory.  |
+|`defaultIfOptionWithoutValue`|The value assigned to the symbol if explicit `null` parameter value is provided by the user or host.|
 |`binding`|The name of the host property to take the value from.|	
-|`replaces`|The text to be replaced by the symbol value in the template files content|	 
+|<a name="replaces"></a>`replaces`|The text to be replaced by the symbol value in the template files content|	 
 |`fileRename`|The portion of template filenames to be replaced by the symbol value.| 
 |`description`|Human readable text describing the meaning of the symbol. This has no effect on template generation.|
-|<a name="isRequired"></a>`isRequired`|Indicates if the parameter is required or not.|
+|<a name="isRequired"></a>`isRequired`|Indicates if the user supplied value is required or not. If set to `true` and user has not specified the value on input, validation error occurs - even if [`defaultValue`](#defaultValue) is present. <br /> Not allowed together with [`isRequiredCondition`](#isRequiredCondition) property. |
 |`choices`|Applicable only when `datatype=choice.`<br />List of available choices. Contains array of the elements: <br />- `choice`: possible value of the symbol.<br />- `description`: human readable text describing the meaning of the choice. This has no effect on template generation. <br /> If not provided, there are no valid choices for the symbol, so it can never be assigned a value.|
 |`allowMultipleValues`|Applicable only when `datatype=choice.`<br />. Enables ability to specify multiple values for single symbol.|
 |<a id="enableQuotelessLiterals"></a>`enableQuotelessLiterals`|Applicable only when `datatype=choice.`<br />. Enables ability to specify choice literals in conditions without quotation.|
 |`onlyIf`| |
 |`forms`|Defines the set of transforms that can be referenced by symbol definitions. Forms allow the specification of a "replaces"/"replacement" pair to also apply to other ways the "replaces" value may have been specified in the source by specifying a transform from the original value of "replaces" in configuration to the one that may be found in the source. [Details](https://github.com/dotnet/templating/wiki/Runnable-Project-Templates---Value-Forms)|
+|<a name="isEnabledCondition"></a>`isEnabledCondition`| Optional condition indicating whether parameter should be processed. More details in [Conditions documentation](Conditions.md#conditional-parameters). |
```

Comment:
Suggestion: would this syntax be better from authoring perspective?
```json
{
    "A":
    {
        "isEnabled" : "true",
        "isRequired": "true"
    },
    "B":
    {
        "isEnabled" : 
        {
            "condition" : "A == true"
        },
        "isRequired":
        {
            "condition": "A == true"
        }
    }
}

```

Or even:
```json
{
    "A":
    {
        "isEnabled" : "true",
        "isRequired": "true"
    },
    "B":
    {
        "isEnabled" : "A == true",
        "isRequired": "A == true"
    }
}
```

Assume `true` is evaluated to `true`.

Reply:
Good suggestions - let me try to to implement that first (`isRequired` being able to be a bool or string), than I'll update this section

Reply:
Fixed in the code and the doc.
I like the new convention much more - thanks for the suggestion!